pkg/didcomm/common/service/did_comm_msg.go

```diff
@@ -107,12 +108,14 @@ func (m DIDCommMsgMap) ThreadID() (string, error) {
 // Metadata returns message metadata
 func (m DIDCommMsgMap) Metadata() map[string]interface{} {
 	if m[jsonMetadata] == nil {
-		return nil
+		m[jsonMetadata] = map[string]interface{}{}
```

Comment:
seems odd to mutate during an accessor.

Reply:
Yeah, but I was trying to avoid another method e.g SetMetadata. I do not think it is good to set metadata in the protocol directly e.g `didmsg[jsonmetadatakey] =map[]interfase{...}`. So I decided that It would be better if didCommmsgMap will always return the map. Actually, there are a few possible ways how to solve this. 
1. Add additional function e.g. SetMetadata
2. Always create an empty map when we use functions like ParseDIDCommMsgMap or NewDIDCommMsgMap
3. The protocol sets metadata to the message (bad idea)

Let me know what do you like more. Suggestions are welcome.

Reply:
I guess I had been thinking about constructor cases (i.e., 2).

Reply:
so updating to 2 then?

Reply:
sure, I will do that

Reply:
Done!